src/DataStructures/ModalVector.hpp

```diff
@@ -92,302 +136,221 @@ class ModalVector {
-    return lhs / _rhs_local;
-  }
+template <typename VT, bool VF>
+bool operator==(const blaze::DenseVector<VT, VF>& lhs,
+                const ModalVector& rhs) noexcept {
+  return ModalVector(lhs) == rhs;
+}
 
-  SPECTRE_ALWAYS_INLINE friend decltype(auto) min(
-      const ModalVector& t) noexcept {
-    return min(t.data_);
-  }
+template <typename VT, bool VF>
+bool operator!=(const blaze::DenseVector<VT, VF>& lhs,
+                const ModalVector& rhs) noexcept {
+  return not(lhs == rhs);
+}
+/// \endcond
 
-  SPECTRE_ALWAYS_INLINE friend decltype(auto) max(
-      const ModalVector& t) noexcept {
-    return max(t.data_);
-  }
+// Specialize the Blaze type traits to correctly handle ModalVector
+namespace blaze {
+template <>
+struct IsVector<ModalVector> : std::true_type {};
 
-  SPECTRE_ALWAYS_INLINE friend decltype(auto) abs(
-      const ModalVector& t) noexcept {
-    return abs(t.data_);
-  }
+template <>
+struct TransposeFlag<ModalVector> : BoolConstant<
+                ModalVector::transpose_flag> {};
 
-  SPECTRE_ALWAYS_INLINE friend decltype(auto) fabs(
-      const ModalVector& t) noexcept {
-    return abs(t.data_);
-  }
-  // @}
+template <>
+struct AddTrait<ModalVector, ModalVector> {
+  using Type = ModalVector;
+};
 
+template <>
+struct AddTrait<ModalVector, double> {
+  using Type = ModalVector;
+};
 
- private:
-  /// \cond HIDDEN_SYMBOLS
-  size_t size_ = 0;
-  InternalStorage_t owned_data_;
-  InternalModalVector_t data_;
-  bool owning_{true};
-  /// \endcond
+template <>
+struct AddTrait<double, ModalVector> {
+  using Type = ModalVector;
 };
 
-/// Output operator for ModalVector
-std::ostream& operator<<(std::ostream& os, const ModalVector& d);
+template <>
+struct SubTrait<ModalVector, ModalVector> {
+  using Type = ModalVector;
+};
 
-/// Equivalence operator for ModalVector
-bool operator==(const ModalVector& lhs, const ModalVector& rhs);
+template <>
+struct SubTrait<ModalVector, double> {
+  using Type = ModalVector;
+};
 
-/// Inequivalence operator for ModalVector
-bool operator!=(const ModalVector& lhs, const ModalVector& rhs);
+template <>
+struct SubTrait<double, ModalVector> {
+  using Type = ModalVector;
+};
+
+template <>
+struct MultTrait<ModalVector, ModalVector> {
+  using Type = ModalVector;
+};
+
+template <>
+struct MultTrait<ModalVector, double> {
+  using Type = ModalVector;
+};
+
+template <>
+struct MultTrait<double, ModalVector> {
+  using Type = ModalVector;
+};
+
+template <>
+struct DivTrait<ModalVector, ModalVector> {
+  using Type = ModalVector;
+};
+
+template <>
+struct DivTrait<ModalVector, double> {
+  using Type = ModalVector;
+};
+
+// Forbid math operations in this specialization of UnaryMap traits for
+// ModalVector that are unlikely to be used on spectral coefficients
+template <typename Operator>
+struct UnaryMapTrait<ModalVector, Operator> {
+  static_assert(not tmpl::list_contains_v<tmpl::list<//blaze::Max, blaze::Min,
+                //blaze::Sqrt,
```

Comment:
why sqrt?

Reply:
also we should list the operations we want, not those we forbid in case new operations are added